cmd/sourced/compose/workdir/factory.go

```diff
@@ -143,78 +136,137 @@ func initialize(dirName string, subPath string, envf envFile) (*Workdir, error)
+		GitbaseVolumeSource: repoDir,
 	}
+	f.addResourceLimits()
+
+	return f
+}
+
+func newOrgEnvFile(dirName string, orgs []string, token string, withForks bool) envFile {
+	f := envFile{
+		ComposeProjectName: fmt.Sprintf("srcd-%s", dirName),
+
+		GitbaseVolumeType:   "volume",
+		GitbaseVolumeSource: "gitbase_repositories",
+		GitbaseSiva:         true,
 
-	noForks := "true"
-	if f.WithForks {
-		noForks = "false"
+		GithubOrganizations: orgs,
+		GithubToken:         token,
+
+		NoForks: !withForks,
 	}
+	f.addResourceLimits()
 
+	return f
+}
+
+func (f *envFile) addResourceLimits() {
 	// limit CPU for containers
-	gitbaseLimitCPU := "0.0"
-	gitcollectorLimitCPU := "0.0"
 	dockerCPUs, err := dockerNumCPU()
 	if err != nil { // show warning
 		fmt.Println(err)
 	}
 	// apply gitbase resource limits only when docker runs without any global limits
 	// it's default behaviour on linux
 	if runtime.NumCPU() == dockerCPUs {
-		gitbaseLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)-0.1)
+		f.GitbaseLimitCPU = float32(dockerCPUs) - 0.1
 	}
+	// always apply gitcollector limit
 	if dockerCPUs > 0 {
-		halfCPUs := float32(dockerCPUs)/2.0
+		halfCPUs := float32(dockerCPUs) / 2.0
+		// let container consume more than a half if there is only one cpu available
+		// otherwise it will be too slow
 		if halfCPUs < 1 {
 			halfCPUs = 1
 		}
-		gitcollectorLimitCPU = fmt.Sprintf("%.1f", halfCPUs-0.1)
+		f.GitcollectorLimitCPU = halfCPUs - 0.1
 	}
 
 	// limit memory for containers
-	gitbaseLimitMem := "0"
 	dockerMem, err := dockerTotalMem()
 	if err != nil { // show warning
 		fmt.Println(err)
 	}
 	// apply memory limits only when only when docker runs without any global limits
 	// it's default behaviour on linux
 	if dockerMem == memory.TotalMemory() {
-		gitbaseLimitMem = strconv.FormatUint(uint64(float64(dockerMem)*0.9), 10)
+		f.GitbaseLimitMem = uint64(float64(dockerMem) * 0.9)
+	}
+}
+
+var newlineChar = "\n"
+
+func init() {
+	if runtime.GOOS == "windows" {
+		newlineChar = "\r\n"
+	}
+}
+
+// implementation can be moved to separate package if we need to marshal any other structs
+// supports only simple types
+func (f envFile) MarshalEnv() ([]byte, error) {
+	var b bytes.Buffer
+
+	v := reflect.ValueOf(f)
+	rType := v.Type()
+
+	for i := 0; i < rType.NumField(); i++ {
+		field := rType.Field(i)
+		fieldEl := v.Field(i)
+		if field.Anonymous {
+			panic("struct composition isn't supported")
+		}
+
+		name := nameToVar(field.Name)
+		switch field.Type.Kind() {
+		case reflect.Slice:
+			slice := make([]string, fieldEl.Len())
+			for i := 0; i < fieldEl.Len(); i++ {
+				slice[i] = fmt.Sprintf("%v", fieldEl.Index(i).Interface())
+			}
+			fmt.Fprintf(&b, "%s=%v%s", name, strings.Join(slice, ","), newlineChar)
+		case reflect.Bool:
+			// marshal false value as empty string instead of "false" string
+			if fieldEl.Interface().(bool) {
+				fmt.Fprintf(&b, "%s=true%s", name, newlineChar)
+			} else {
+				fmt.Fprintf(&b, "%s=%s", name, newlineChar)
```

Comment:
If we consider `false` is the zero value and should be left to the default value in the docker compose file, I think this line should not be included (leaving `name` unset instead of set to an empty string).

Since https://github.com/src-d/sourced-ce/pull/212 we are using `${VARIABLE-default}` instead of `${VARIABLE:-default}`. From [the docs](https://docs.docker.com/compose/compose-file/#variable-substitution):

> `${VARIABLE:-default}` evaluates to default if VARIABLE is unset or empty in the environment.
> `${VARIABLE-default}` evaluates to default only if VARIABLE is unset in the environment.

Reply:
Hm. I see your point but we always write all the values. For example `GITBASE_LIMIT_MEM` on my machine is presented in `.env` though it's not used as I use macOs. It would be strange to do such inconsistency. I don't mind to no-write empty values at all if you prefer. But imo current way is more clear. I can see what is set and not set by looking only in `.env` instead of inspecting source code.

Reply:
So this means for `sourced-ce` we don't really support the default values in the `docker-compose.yml` file, since we are always going to provide a value.

Now I'm wondering if we want to have default values in the compose file at all. Because now the only reason for them is if someone downloads the compose file and uses it individually outside of the `sourced-ce` context.
Which I guess is fine, if it does not affect `sourced-ce` logic in any way...

Reply:
> So this means for sourced-ce we don't really support the default values in the docker-compose.yml file, since we are always going to provide a value.

yes

> Now I'm wondering if we want to have default values in the compose file at all. 

we use `docker-compose.yml` with just `docker-compose` and not our binary in demos. So it's nice to have default values.

Reply:
> I can see what is set and not set by looking only in .env instead of inspecting source code.

I strongly agree with this 👍, it's really convenient.